assets/app/features/upgrade/controllers/upgrade-nodes.controller.js

```diff
@@ -58,17 +73,44 @@
         function beginUpgradeNodes() {
             vm.nodesUpgrade.running = true;
 
-            upgradeFactory.upgradeNodes()
+            if (vm.nodesUpgrade.subStep === 'controller_nodes' && vm.nodesUpgrade.subStepStatus === 'finished') {
```

Comment:
Please declare those variables in `vm.nodesUpgrade`.

Reply:
Will do.